### ISSUE TITLE
Autocomplete: consider to use `aria-posinset` and `aria-setsize`

### ISSUE THREAD

**Issue by afercia** [alphagov/accessible-autocomplete#277]
Testing a bit the autocomplete with Firefox and NVDA, it works pretty well (maybe it's the browser / screen reader combo where it works best) and it's great to see all the work that has been done here.

I've noticed something that could be improved when the suggestions list gets filtered. Seems Firefox and NVDA are not able to calculate and announce the new length of the list. In this case, I _think_ `aria-posinset` and `aria-setsize` could help.

In the screenshot below, when the list shows just 2 results, Firefox and NVDA still announce the initial length of the list (which was 165), thus the two options are announced as "Italy 1 of 165"  and "Mauritania 2 of 165", while they should be announced as "Italy 1 of 2"  and "Mauritania 2 of 2":

![screenshot 27](https://user-images.githubusercontent.com/1682452/42232772-c3c17e26-7eef-11e8-9920-ca4e1ce4bc27.png)

I haven't looked at the code, but if this was an array, I guess there's no interest in communicating the original position of the option within the initial 165 options so I guess [aria-posinset](https://www.w3.org/TR/wai-aria-1.1/#aria-posinset) should be the array item index and [aria-setisize](https://www.w3.org/TR/wai-aria-1.1/#aria-setsize) the array length.

Worth noting both properties are mentioned for the Listbox widget in the ARIA Authoring Practices: https://www.w3.org/TR/wai-aria-practices/#Listbox and allowed for the `option` role: https://www.w3.org/TR/wai-aria-1.1/#option

**Comment by nickcolley** [comment]
Will be in the 2.0.0 release :)
